segmentation/utils/signal_extraction.py

```diff
@@ -1,31 +1,95 @@
 import numpy as np
 
 
-def simple_weighted_extraction(cell_coords, image_data):
+# TODO: work on weighted extraction and implement other discussed techniques of extraction
+def positive_pixels_extraction(cell_coords, image_data):
     """
-    Define a weighted extraction technique based on how close the coordinate is to the center
+    Extract channel counts by summing over the number of non-zero pixels in the cell,
+    improves on default_extraction by not distinguishing between pixel expression values
 
     Args:
-        cell_coords: tuples of values representing pixels within one cell
-        image_data: array containing channel counts
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
 
     Returns:
-        np.array: sum of counts for each channel
+        channel_counts (numpy): sum of counts for each channel
     """
 
-    pass
+    # transpose coords so they can be used to index an array
+    cell_coords = cell_coords.T
+    channel_values = image_data.values[tuple(cell_coords)]
+
+    # sum up based on a binary mask that is 1 if the expression value > 0 else 0
+    channel_counts = np.sum(channel_values > 0, axis=0)
+
+    return channel_counts
+
+
+def center_weighting_extraction(cell_coords, image_data):
```

Comment:
Currently this doesn't work as intended. Take a look at the function that is calling signal_extraction: https://github.com/angelolab/segmentation/blob/af92d9cd0dbc299279005c059f5a183222183b7f/segmentation/utils/segmentation_utils.py#L441

It first identifies the coordinates of all pixels belonging to the current cell, and then passes those coordinates, along with the image data of the entire FOV, to this helper function. 

Picking the center of this image won't work. Instead, we need to compute the center of the cell. This isn't possible to do just from the list of cell_coords. Instead, you'll need to use regionprops_table to generate a list of cell_centers, and use that for this seed. 
My guess is that for this function, we'll need to pass in the segmentation_label mask corresponding just to that cell, rather than a list of coordinates.

Reply:
Ok got it, something seemed off about indexing image_data but for some reason couldn't wrap my head around it haha.